CHANGELOG.md

```diff
@@ -2,6 +2,7 @@ Unreleased
 ----------
 * Fixes a bug with `EnsureAuthenticatedLinks` causing deep links to not work [#1549](https://github.com/Shopify/shopify_app/pull/1549)
 * Ensure online token is properly used when using `current_shopify_session` [#1566](https://github.com/Shopify/shopify_app/pull/1566)
+* Detect the use of incompatible controller concerns [#1560](https://github.com/Shopify/shopify_app/pull/1560). Depending on how your app is structured, this may result in an error on startup even though the app was working correctly. However, it is helping exposing an existing flaw that may have caused issues in future.
```

Comment:
Makes sense, I'll try switch it over to a deprecation, e.g.:

> We detected the use of incompatible concerns (`RequireKnownShop` and `LoginProtection`) in a controller, which may lead to unpredictable behavior. In a future release of this library this will raise an error.

Reply:
Can we easily log the controller name to make debugging it easier?

Reply:
The logs _should_ already indicate the location, but I've added it just in case. Here's how it looks now:

```
DEPRECATION WARNING: We detected the use of incompatible concerns (RequireKnownShop and LoginProtection) in ProductsController,
which may lead to unpredictable behavior. In a future release of this library this will raise an error.
 (called from include at /Users/andyw8/src/github.com/Shopify/shopify_app/my_shopify_app_clash/app/controllers/products_controller.rb:5)
```